### ISSUE TITLE
mlflow.exceptions.RestException: BAD_REQUEST

### ISSUE THREAD

**Issue by tamilselvan1588** [mlflow/mlflow#13436]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Other

### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### MLflow version

mlflow==1.30.0

### System information

- **OS Platform and Distribution - Ubuntu 22.04.5 LTS
- **Python version**: Python 3.8
- 

### Describe the problem

I am getting the below error while running the code. Can you please help to resolve the error.

mlflow.exceptions.RestException: BAD_REQUEST: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "tag_pk"

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

```
REPLACE_ME
```

### Stack trace

```
REPLACE_ME
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [x] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [x] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by daniellok-db** [comment]
@tamilselvan1588 could you provide a code example to reproduce the error? Also, it seems your version of MLflow is very old. Have you tried upgrading?

**Comment by tamilselvan1588** [comment]
@daniellok-db, i tried with the mlflow 2.17.0 version but it is also giving the same error.

Full error of the issue.
mlflow.exceptions.RestException: BAD_REQUEST: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "tag_pk"
DETAIL:  Key (key, run_uuid)=(mlflow.runName, ******) already exists.

**Comment by daniellok-db** [comment]
Could you provide the code and a description of what you're trying to do?

**Comment by tamilselvan1588** [comment]
import shutil
from pathlib import Path
import mlflow
from sklearn.linear_model import LogisticRegression
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
from sklearn.linear_model import LogisticRegression

artifact_dir = ".models"
model_path = Path("model_test")

run_name = "test-run"
experiment_name = "test-experiment"

if model_path.exists():
    shutil.rmtree(model_path)
model_path.mkdir(exist_ok=False, parents=True)

X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

mlflow.set_experiment(experiment_name)

with mlflow.start_run(run_name=run_name):
    model = LogisticRegression().fit(X_train, y_train)
    y_pred = model.predict(X_test)
    accuracy= accuracy_score(y_test, y_pred)

    mlflow.sklearn.save_model(model, model_path)
    mlflow.log_params(model.get_params())
    mlflow.log_metric("accuracy", accuracy)
    
    
    When specifying a run name in with mlflow.start_run(run_name=run_name), the key containing mlflow.runName and run_uuid is created twice, leading to the following exception:
mlflow.exceptions.RestException: BAD_REQUEST: (psycopg2.errors.UniqueViolation) duplicate key value violates unique constraint "tag_pk" DETAIL:  Key (key, run_uuid)=(mlflow.runName, 64e68fdf8d6a489b920dc875094e1d0b) already exists.

Please see the stack trace for more details. This occurs for version 1.30.1 and  2.17.0

**Comment by daniellok-db** [comment]
Could you try without specifying the run name and see if the error still occurs?

**Comment by tamilselvan1588** [comment]
Thanks, the issue is resolved after removing the run name from code.
